internal/controllers/rewards.go

```diff
@@ -122,68 +127,85 @@ func (r *RewardsController) GetUserRewards(c *fiber.Ctx) error {
 					return opaqueInternalError
 				}
 
-				if slices.Contains(ints, vendorToID["AutoPi"]) {
+				if slices.Contains(ints, vendorToID["AutoPi"].ID) {
 					uriAP := UserResponseIntegration{
-						ID:                   vendorToID["AutoPi"],
+						ID:                   vendorToID["AutoPi"].ID,
 						Vendor:               "AutoPi",
 						OnChainPairingStatus: "Unpaired",
 						DataThisWeek:         true,
 					}
 
-					if device.AftermarketDeviceTokenId != nil {
+					if device.AftermarketDevice.TokenId != 0 {
```

Comment:
I feel like you have a huge chance of hitting nils in this file. Let's pair on this today.